### PR TITLE
On board 1-click release for OpenSearch Migrations

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
On board 1-click release for OpenSearch Migrations

### Issues Resolved
Part of: https://github.com/opensearch-project/opensearch-build/issues/4085

Is this a backport? If so, please add backport PR # and/or commits #

### CODE REVIEW

File: .github/workflows/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          issue-title: 'Release opensearch-migrations version ${{ steps.get_data.outputs.version }}'
+          issue-body: "Please approve or deny the release of opensearch-migrations **TAG**: ${{ github.ref_name }}  **COMMIT**: ${{ github.sha }} **VERSION** : ${{ steps.get_data.outputs.version }} "
+          exclude-workflow-initiator-as-approver: true
+
+      - name: Draft a release
+        uses: softprops/action-gh-release@v1
+        with:
+          draft: true
+          generate_release_notes: true
+          files: artifacts.tar.gz
```
Comment (gaiksaya) [inline .github/workflows/release-drafter.yml:33]: We need to first create this tar.gz before uploading. Can you please try out this workflow in your fork?
Thanks!

Comment (prudhvigodithi) [reply .github/workflows/release-drafter.yml:33]: Sure Sayali, will test it.

Comment (prudhvigodithi) [reply .github/workflows/release-drafter.yml:35]: Hey @gaiksaya with the workflow I have able to create a draft release upon a tag (0.3.0) push.
Screenshot 2023-10-04 at 1 59 28 PM

File: jenkins/release.jenkinsFile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+lib = library(identifier: 'jenkins@5.9.1', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/opensearch-project/opensearch-build-libraries.git',
+]))
+
+standardReleasePipelineWithGenericTrigger(
+    overrideDockerImage: 'opensearchstaging/ci-runner:release-centos7-clients-v4',
```
Comment (gaiksaya) [inline jenkins/release.jenkinsFile:7]: No need to override, the 5.9.1 version has this by default: https://github.com/opensearch-project/opensearch-build-libraries/blob/5.9.1/vars/standardReleasePipelineWithGenericTrigger.groovy#L31

**Review by gaiksaya — APPROVED** [review]
Minor nit about overriding image. Other than that LGTM!

File: jenkins/release.jenkinsFile (hunk 2 above)
Comment (prudhvigodithi) [reply jenkins/release.jenkinsFile]: Updated.

**Comment by prudhvigodithi** [comment]
Need one maintainers approval @chelma @gregschohn @kartg @lewijacn @mikaylathompson @okhasawn @sumobrian please check. Thanks

File: README.md [hunk 3]
```diff
@@ -60,3 +60,13 @@ See [CONTRIBUTING](CONTRIBUTING.md#security-issue-notifications) for more inform
...
 This project is licensed under the Apache-2.0 License.
+
+
+## Releasing
+
+The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [maintainers](MAINTAINERS.md).
+
+1. Create a tag, e.g. 0.1.0, and push it to this GitHub repository.
+2. The [release-drafter.yml](.github/workflows/release-drafter.yml) will be automatically kicked off and a draft release will be created.
+3. This draft release triggers the [jenkins release workflow](https://build.ci.opensearch.org/job/opensearch-migrations-release) as a result of which the opensearch-migrations toolset is released and published on artifacts (https://artifacts.opensearch.org/migrations/0.1.0/opensearch-migrations-0.1.0.tar.gz).
```
Comment (gaiksaya) [inline README.md:71]:
```suggestion
3. This draft release triggers the [jenkins release workflow](https://build.ci.opensearch.org/job/opensearch-migrations-release) as a result of which the opensearch-migrations toolset is released and published on artifacts.opensearch.org (https://artifacts.opensearch.org/migrations/0.1.0/opensearch-migrations-0.1.0.tar.gz). 
```

**Review by sumobrian — APPROVED** [review]

**Pull request merged**